### PR TITLE
Product Visualization and Management

### PR DESCRIPTION
**Pull request by cristianbyte**
### Enhance Product Visualization and Management 

This pull request addresses the user stories related to enhancing product visualization and inventory management in the store application. It introduces features to display product categories and locations, as well as functionalities to identify demanded products and manage low stock items efficiently.

**Changes Implemented:**

1. For Viewing Product Categories and Locations:
    -   Implemented a feature to display the categories and exact locations of all products.
    -   Added filtering options by product category to facilitate product search and browsing.
    -   Enabled functionality for easy updates to product locations through an intuitive interface.
 2. For Identifying Demanded Products and Low Stock Items:
      - A list of best-selling products based on sales data.
      - Products with low stock levels and alerts for those nearing their expiration date.
      - Configuration options for setting low stock thresholds to trigger proactive alerts.

### CODE REVIEW

File: src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/request/update/ProductUpdateLocationResponse.java
Comment (SMoreno73) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/request/update/ProductUpdateLocationResponse.java:1]: Why this response is empty?

File: src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/response/BestSellingResponse.java
Comment (SMoreno73) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/response/BestSellingResponse.java:1]: Do you need @ToString?, what do you do?

File: src/main/java/com/riwi/localstorage/riwi_local_storage/domain/entities/Product.java
Comment (SMoreno73) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/domain/entities/Product.java:1]: Why do you need @ToString?

File: src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/mappers/BranchMapper.java
Comment (SMoreno73) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/mappers/BranchMapper.java:1]: I suggest you put all @Mapping in a agrupation notation -> @Mapppings

File: src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/CustomService.java
Comment (SMoreno73) [inline src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/CustomService.java:1]: Is a good have a pattern, here you forget the final

**Review by SMoreno73 — CHANGES_REQUESTED** [review]
Fix or explan me the reason why you do that in that way

File: src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/response/BestSellingResponse.java
Comment (cristianbyte) [reply src/main/java/com/riwi/localstorage/riwi_local_storage/api/dto/response/BestSellingResponse.java:1]: It was for testing, I'm going to remove it.

**Review by SMoreno73 — APPROVED** [review]

**Pull request merged**